src/main/ImGuiZepEditor.cpp

```diff
@@ -511,4 +523,75 @@ void ImGuiZepEditor::Render(const char* id, const ImVec2& displaySize)
 	ImGui::EndChild();
 }
 
+void ImGuiZepEditor::Clear()
+{
+	m_buffer->Clear();
+}
+
+void ImGuiZepEditor::GetText(std::string& outBuffer) const
+{
+	outBuffer.reserve(GetTextLength());
+	auto& workingBuffer = m_buffer->GetWorkingBuffer();
+	std::copy(workingBuffer.begin(), workingBuffer.end(), std::back_inserter(outBuffer));
+	// copy 1 less character to avoid adding the null terminator into the std::string size.
+	outBuffer.pop_back();
+}
+
+size_t ImGuiZepEditor::GetTextLength() const
+{
+	// buffer size includes the null terminator the text length should not.
+	return m_buffer->GetWorkingBuffer().size() - 1;
+}
+
+void ImGuiZepEditor::SetSyntaxProvider(const char* syntaxName) const
+{
+	GetEditor().SetBufferSyntax(*m_buffer, syntaxName);
+}
+
+Zep::GlyphIterator ImGuiZepEditor::InsertText(Zep::GlyphIterator position, std::string_view text, ImU32 color /*= -1*/)
+{
+	Zep::ChangeRecord changeRecord;
+	m_buffer->Insert(position, text, changeRecord);
+
+	return position.Move(static_cast<long>(text.length()));
+}
+
+void ImGuiZepEditor::SetText(std::string_view text, bool appendNewLine /* = false */)
+{
+	Clear();
+
+	Zep::GlyphIterator cursor = m_window->GetBufferCursor();
+	bool cursorAtEnd = m_window->IsAtBottom();
+
+	std::string_view lineView = text;
+
+	std::vector<ImU32> colorStack;
+
+	InsertText(m_buffer->End(), lineView);
+
+	if (appendNewLine)
+		InsertText(m_buffer->End(), "\n");
+}
+
+void ImGuiZepEditor::SetFontSize(int size)
+{
+	m_fontSize = size;
+	SetFont(Zep::ZepTextType::Text, mq::imgui::ConsoleFont, m_fontSize);
+}
+
+int ImGuiZepEditor::GetWindowFlags() const
+{ 
+	return m_window->GetWindowFlags(); 
+}
+
+void ImGuiZepEditor::SetWindowFlags(int flags) const
+{
+	m_window->SetWindowFlags(flags);
+}
+
+std::shared_ptr<mq::imgui::ImGuiZepEditor> ImGuiZepEditor::Create(std::string_view id)
+{
+	return std::make_shared<ImGuiZepEditor>(id);
```

Comment:
The constructor isn't private, why do this? It makes it seem like there's more to creating the editor than just make_shared